dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/CacheListener.java

```diff
@@ -89,7 +89,8 @@ public void dataChanged(String path, Object value, EventType eventType) {
         // TODO We limit the notification of config changes to a specific path level, for example
         //  /dubbo/config/service/configurators, other config changes not in this level will not get notified,
         //  say /dubbo/config/dubbo.properties
-        if (path.split("/").length >= 5) {
+        final int MIN_PATH_ARRAY_LENGTH = 5;
```

Comment:
It is better to make it static.

Reply:
It is better to be local, because it is only used in one method.

Reply:
No, it is not. As a local constant, it will be allocated every time the method is invoked. I don't think it should be local.

Reply:
I make an experiment.
`public class Test {
    private static final int CLZ_FIELD = 65535;
    public static void main(String[] args) {
        final int localVariable = 65536;
        System.out.println(localVariable);
        System.out.println(CLZ_FIELD);
    }
}`
After decompilation， I got
`  public static void main(java.lang.String[]);
    Code:
       0: ldc           #2                  // int 65536
       2: istore_1
       3: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
       6: ldc           #2                  // int 65536
       8: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
      11: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
      14: ldc           #6                  // int 65535
      16: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
      19: return`
Every time the local constant need one more step ('0: ldc           #2') and one more stack space than static constant. 
So you are right,  I will make it static.

Reply:
I make an experiment.
`
public class Test {
    private static final int CLZ_FIELD = 65535;
    public static void main(String[] args) {
        final int localVariable = 65536;
        System.out.println(localVariable);
        System.out.println(CLZ_FIELD);
    }
}
`
After decompilation， I got
`
  public static void main(java.lang.String[]);
    Code:
       0: ldc           #2                  // int 65536
       2: istore_1
       3: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
       6: ldc           #2                  // int 65536
       8: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
      11: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
      14: ldc           #6                  // int 65535
      16: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
      19: return
`
Every time the local constant need one more step ('0: ldc #2') and one more stack space than static constant.
So you are right, I will make it static.

Reply:
I make an experiment.

public class Test {
    private static final int CLZ_FIELD = 65535;
    public static void main(String[] args) {
        final int localVariable = 65536;
        System.out.println(localVariable);
        System.out.println(CLZ_FIELD);
    }
}

After decompilation， I got

  public static void main(java.lang.String[]);
    Code:
       0: ldc           #2                  // int 65536
       2: istore_1
       3: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
       6: ldc           #2                  // int 65536
       8: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
      11: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
      14: ldc           #6                  // int 65535
      16: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
      19: return

Every time the local constant need one more step ('0: ldc #2') and one more stack space than static constant.
So you are right, I will make it static.

Reply:
I make an experiment.

`
public class Test {
    private static final int CLZ_FIELD = 65535;
    public static void main(String[] args) {
        final int localVariable = 65536;
        System.out.println(localVariable);
        System.out.println(CLZ_FIELD);
    }
}

`

After decompilation， I got

`
public static void main(java.lang.String[]);
  Code:
     0: ldc           #2                  // int 65536
     2: istore_1
     3: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
     6: ldc           #2                  // int 65536
     8: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
    11: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
    14: ldc           #6                  // int 65535
    16: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
    19: return
`

Every time the local constant need one more step ('0: ldc #2') and one more stack space than static constant.
So you are right, I will make it static.

Reply:
I make an experiment.

```
public class Test {
    private static final int CLZ_FIELD = 65535;
    public static void main(String[] args) {
        final int localVariable = 65536;
        System.out.println(localVariable);
        System.out.println(CLZ_FIELD);
    }
}
```

After decompilation， I got


public static void main(java.lang.String[]);
```
  Code:
     0: ldc           #2                  // int 65536
     2: istore_1
     3: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
     6: ldc           #2                  // int 65536
     8: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
    11: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
    14: ldc           #6                  // int 65535
    16: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
    19: return
```

Every time the local constant need one more step ('`0: ldc #2`') and one more stack space than static constant.
So you are right, I will make it static.

Reply:
I make an experiment.
```
public class Test {
    private static final int CLZ_FIELD = 65535;
    public static void main(String[] args) {
        final int localVariable = 65536;
        System.out.println(localVariable);
        System.out.println(CLZ_FIELD);
    }
}
```
After decompilation， I got
```
public static void main(java.lang.String[]);
  Code:
     0: ldc           #2                  // int 65536
     2: istore_1
     3: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
     6: ldc           #2                  // int 65536
     8: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
    11: getstatic     #3                  // Field java/lang/System.out:Ljava/io/PrintStream;
    14: ldc           #6                  // int 65535
    16: invokevirtual #4                  // Method java/io/PrintStream.println:(I)V
    19: return
```
Every time the local constant need one more step ('`0: ldc #2`') and one more stack space than static constant.
So you are right, I will make it static.